src/charm.py

```diff
@@ -418,6 +446,55 @@ def _on_tls_certificate_pki_certificate_available(self, event: CertificateAvaila
         """Handle the tls-certificates-pki certificate available event."""
         self._add_ca_certificate_to_pki_secrets_engine()
 
+    def _on_vault_pki_certificate_creation_request(
+        self, event: CertificateCreationRequestEvent
+    ) -> None:
+        """Handle the vault-pki certificate creation request event."""
+        self._generate_pki_certificate_for_requirer(
+            event.certificate_signing_request, event.relation_id
+        )
+
+    def _generate_pki_certificate_for_requirer(self, csr: str, relation_id: int):
+        """Generate a PKI certificate for a TLS requirer."""
+        if not self.unit.is_leader():
+            logger.debug("Only leader unit can handle a vault-pki certificate request")
+            return
+        if not self._is_peer_relation_created():
+            logger.debug("Peer relation not created")
+            return
+        vault = self._get_initialized_vault_client()
+        if not vault:
+            logger.debug("Failed to get initialized Vault")
+            return
+        common_name = self._get_config_common_name()
+        if not common_name:
+            logger.error("Common name is not set in the charm config")
+            return
+        if not vault.is_pki_role_created(role=PKI_ROLE, mount=PKI_MOUNT):
+            logger.debug("PKI role not created")
+            return
+        requested_csr = csr
+        requested_common_name = get_common_name_from_csr(requested_csr)
+        try:
+            certificate = vault.sign_pki_certificate_signing_request(
+                mount=PKI_MOUNT,
+                role=PKI_ROLE,
+                csr=requested_csr,
+                common_name=requested_common_name,
+            )
+        except VaultClientError:
+            logger.warning(
+                "Failed to sign the PKI certificate signing request for %s", requested_common_name
+            )
+            return
```

Comment:
I don't think having 2 excepts for 1 call makes sense. One of the reasons we have the library is so we don't have to bubble up every error hvac produces. We should just log and return None from the client when the csr can't be signed and return if not here.

Reply:
I agree that we probably don't want to bubble up every error hvac produces. In this context however, we tried to do something and it failed so I felt raising was the right approach. 

I'm ok with returning None instead.

Reply:
Done